qiskit/passmanager/passmanager.py

```diff
@@ -211,12 +216,25 @@ def callback_func(**kwargs):
                 execution is enabled. This argument overrides ``num_processes`` in the user
                 configuration file, and the ``QISKIT_NUM_PROCS`` environment variable. If set
                 to ``None`` the system default or local user configuration will be used.
-
+            property_set: Currently a key that will be interpreted as all other arbitrary
+                ``kwargs``.  In Qiskit 2.0, this will instead seed the :class:`.PropertySet` of the
+                compilation (but does not do so in this version).
             kwargs: Arbitrary arguments passed to the compiler frontend and backend.
 
         Returns:
             The transformed program(s).
         """
+        if property_set is not _NOT_PRESENT:
+            warnings.warn(
+                "From Qiskit 2.0, 'property_set' will be a reserved keyword argument of"
+                " 'BasePassManager.run', and not passed on to the conversion functions."
+                " This subclass of 'BasePassManager' needs to use a different keyword argument"
+                " for passing on this data.",
+                FutureWarning,
+                stacklevel=2,
+            )
+            kwargs["property_set"] = property_set
```

Comment:
I wonder if this indentation was an oversight? Shouldn't we populate the kwargs with `property_set` when it's present too? At least this is what I understood from the docstring. I might be wrong.

```suggestion
        kwargs["property_set"] = property_set
```

Reply:
If the `property_set` parameter has the sentinel value `_NOT_PRESENT`, it means that the function was called without setting that kwarg - there's no valid way for a user to have passed the special sentinel object other than by not typing the argument.  So if the conditional fails (i.e. `property_set is _NOT_PRESENT`) then it shouldn't be passed on, because the user didn't give us one.

Reply:
hah, so I was wrong. Glad I asked :)

Reply:
It took me 3 re-reads but I got why, I was not reading the `not` (it was "not not present", not "not present"), which was what confused me in the first place. Anyways, reading comprehension, now I get it.

Reply:
oh yeah, the naming isn't super ideal.  Do you think `_MISSING` or `_ABSENT` would make it clearer?  There's still _kind of_ a double negative in there (asserting lack of absence), but might be easier to read?